### ISSUE TITLE
Actuarial: InsuranceProgram.process_claim adds uncovered excess loss to deductible_paid — overstates insured's retained loss

### ISSUE THREAD

**Issue by AlexFiliakov** [AlexFiliakov/Ergodic-Insurance-Limits#1069]
## Problem Statement

In `insurance_program.py`, the `process_claim()` method (line 621) has logic at lines 679-684 that adds uncovered loss (loss above all layer limits) back into `deductible_paid`:

```python
if total_covered < claim_amount:
    uncovered_loss = claim_amount - total_covered
    deductible_paid += uncovered_loss  # Line 684
```

This conflates three distinct insured retention components:
1. **Deductible/SIR**: The planned self-insured retention below the first layer
2. **Gaps**: Losses falling in gaps between layers (if any)
3. **Excess of limits**: Losses above the top of the insurance tower

By adding uncovered excess into `deductible_paid`, the returned result makes it appear that the insured's deductible was larger than configured, which:
- Misrepresents the insurance program's structure
- Makes loss analytics confusing (deductible analysis shows amounts above program limits)
- Prevents proper allocation between planned retention vs. uninsured excess

Note: This is related to existing issue #840 but focuses on a different aspect (the accounting of uncovered loss, not the deductible/SIR conflation).

## v1.0 Impact
All downstream analytics (ROE calculations, cost allocation, experience rating) that use `deductible_paid` will be contaminated with uninsured excess amounts.

## Affected Code
- `ergodic_insurance/insurance_program.py` lines 679-684 (uncovered loss addition in `process_claim`)

## Current Behavior
`ClaimResult.deductible_paid` includes both the actual deductible AND any loss above the insurance tower limits.

## Expected Behavior (Standard industry claim accounting)
The claim result should separately track:
- `deductible_paid`: Losses within the deductible/SIR only
- `uncovered_loss`: Losses above the top of the insurance tower
- `gap_loss`: Losses falling in gaps between layers (if applicable)
- `total_retained`: Sum of all retained components

## Alternative Solutions
1. **Add separate fields** to `ClaimResult` for gap loss vs. excess-of-limits loss.
2. **Stop adding uncovered loss to deductible_paid** — keep it only in `uncovered_loss` field.
3. **Add a `total_retained` field** that sums deductible + uncovered, but keep them distinct.

## Recommended Approach
Option 2: Remove the line that adds uncovered loss to `deductible_paid`. The `uncovered_loss` field already captures this correctly. Users who need total retained can sum `deductible_paid + uncovered_loss`.

## Acceptance Criteria
- [ ] `deductible_paid` reflects only the deductible/SIR amount
- [ ] `uncovered_loss` correctly reflects excess-of-limits loss
- [ ] `total_claim = deductible_paid + insurance_recovery + uncovered_loss` identity holds
- [ ] Downstream analytics correctly separate retention components
